fip-x.md

```diff
@@ -0,0 +1,115 @@
+
+![Image of UML Diagram](https://github.com/iwill610/FDM_UMLsd/blob/main/FIO%20Domain%20Marketplace.png)
+
+This diagram reads left to right and top to bottom.  There are three different scenarios outlined and they are distinguished by the color of the arrows.  The first scenario, in red, starts will the user using the website to place a domain up for sale for a specified price.  This user is denoted as "seller."  The website then communicates that the smart contract needs to use the "list" action, which will transfer ownership of the address to the contract account and place that address for sale on the website.  
+
+The second scenario, in blue, starts will the seller cancelling their listing on the website.  The website then communicates with the smart contract in order to transfer ownership of the domain back to the seller.  
+
+The last scenario, in green, starts with a different user buying a listing on the website.  This second user is denoted as "buyer."  The website then communicates with the smart contract in order to transfer ownership of the address to the buyer and transfer the funds used to buy the domain to the seller. 
+
+## New actions
+### List
+This action will post the domain up for sale on the marketplace for a specified price. 
+#### New action: *list*
+
+#### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|price|Yes|Positive Int|The amount of FIO for which the seller wants to sell their domain|
+|fio_public_key|Yes|FIO Public Key|Valid FIO Public Key of the seller|
+|fio_domain|Yes|FIO Domain|Valid and unexpired FIO Domain|
+
+##### Example
+```
+{
+	"price": 20,
+	"fio_public_key": FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj,
+	"fio_domain": "alice"
+}
+```
+
+### Cancel Listing
+This action will take the domain off of the marketplace and return ownership of the domain to the seller. 
+#### New action: *cancel_listing*
+
+#### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_public_key|Yes|FIO Public Key|Valid FIO Public Key of the person who is trying to cancel the listing|
+|fio_domain|Yes|FIO Domain|Valid and unexpired FIO Domain|
+
+##### Example
+```
+{
+	"fio_public_key": FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj,
+	"fio_domain": "alice"
+}
+```
+
+### Buy Listing
+This action will post the domain up for sale on the marketplace for a specified price. 
+#### New action: *buy_listing*
+
+#### Request
+Describe what parameters are passed in. Example:
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|cost|Yes|Positive Int|The amount of FIO the buyer inputs to purchase a domain|
+|fio_public_key|Yes|FIO Public Key|Valid FIO Public Key of the buyer|
+|fio_domain|Yes|FIO Domain|Valid and unexpired FIO Domain|
+##### Example
+```
+{
+	"cost": 20,
+	"fio_public_key": FIO8PRe4WRZJj5mkem6qVGKyvNFgPsNnjNN6kPhh6EaCpzCVin5Jj
+	"fio_domain": "alice"
+}
+```
+
```

Comment:
Per convention, Rationale, Implementation, Backwards Compatibility, should be added

Reply:
I intend for all these to be included once we get further along in the process.  They were blank or had very little in the copy that I have so I did not include them in this version of the draft that you are seeing now.

Reply:
I would add as placeholders